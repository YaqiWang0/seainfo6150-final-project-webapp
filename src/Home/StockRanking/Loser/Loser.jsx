import React, { useEffect, useState} from "react"
import RankingItem from "../RankingItem";
import style from "../StockRanking.module.css"

const Loser = (props) => {

    const loser = []
    for (var i = 0; i < 5; i++) {
        loser.push(props.data[i])
    }
    return (
        <div  className={style.loser}>
            <div className={style.title}> Most Loser Stock Today</div>
            <div className={style.list}>
            {loser.map((item) => (
                <div>
                <RankingItem isGainer = {false} stock = {item} key = {item.ticker} login = {props.login}/>
                <hr className={style.line}/>
                </div>
            ))}
            </div>
        </div>
    )
}

export default Loser