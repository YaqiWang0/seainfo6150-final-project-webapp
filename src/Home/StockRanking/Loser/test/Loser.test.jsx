import React from "react"
import data from "./data";
import Loser from "../Loser";


describe("Loser test", () => {
    const login = {
        login: true,
        userId: "12345",
        email: "test@test.com"

    }
    it('when the item is not clicked', function () {
        const {container} = render(<Loser data={Object.values(data)} login = {login}/>);
        expect(container).toMatchSnapshot();
    })
})
