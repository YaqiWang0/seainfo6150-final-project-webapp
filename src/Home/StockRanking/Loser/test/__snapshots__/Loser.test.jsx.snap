// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loser test when the item is not clicked 1`] = `
<div>
  <div
    class="loser"
  >
    <div
      class="title"
    >
       Most Loser Stock Today
    </div>
    <div
      class="list"
    >
      <div>
        <div
          class="ranking_item"
          role="click"
        >
          <div
            class="name"
          >
            <span
              class="stock_name"
            >
              SNOA
            </span>
            <span
              class="company_name"
            >
              Sonoma Pharmaceuticals Inc
            </span>
          </div>
          <span
            class="loser_price_text"
          >
            $10.75
          </span>
        </div>
        <hr
          class="line"
        />
      </div>
      <div>
        <div
          class="ranking_item"
          role="click"
        >
          <div
            class="name"
          >
            <span
              class="stock_name"
            >
              SLDB
            </span>
            <span
              class="company_name"
            >
              Solid Biosciences Inc
            </span>
          </div>
          <span
            class="loser_price_text"
          >
            $5.7706
          </span>
        </div>
        <hr
          class="line"
        />
      </div>
      <div>
        <div
          class="ranking_item"
          role="click"
        >
          <div
            class="name"
          >
            <span
              class="stock_name"
            >
              EQS
            </span>
            <span
              class="company_name"
            >
              EQUUS Total Return Inc
            </span>
          </div>
          <span
            class="loser_price_text"
          >
            $1.755
          </span>
        </div>
        <hr
          class="line"
        />
      </div>
      <div>
        <div
          class="ranking_item"
          role="click"
        >
          <div
            class="name"
          >
            <span
              class="stock_name"
            >
              NH
            </span>
            <span
              class="company_name"
            >
              NantHealth Inc
            </span>
          </div>
          <span
            class="loser_price_text"
          >
            $4.18
          </span>
        </div>
        <hr
          class="line"
        />
      </div>
      <div>
        <div
          class="ranking_item"
          role="click"
        >
          <div
            class="name"
          >
            <span
              class="stock_name"
            >
              VTVT
            </span>
            <span
              class="company_name"
            >
              vTv Therapeutics Inc
            </span>
          </div>
          <span
            class="loser_price_text"
          >
            $3.2
          </span>
        </div>
        <hr
          class="line"
        />
      </div>
    </div>
  </div>
</div>
`;
